apis/apps/v1alpha1/cluster_types.go

```diff
@@ -278,31 +287,55 @@ type ClusterStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
+// ShardingSpec defines the sharding spec.
+type ShardingSpec struct {
+	// name defines sharding template name, this name is also part of Service DNS name, so this name will comply with IANA Service Naming rule.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=22
+	// +kubebuilder:validation:Pattern:=`^[a-z]([a-z0-9\-]*[a-z0-9])?$`
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="name is immutable"
+	Name string `json:"name"`
+
+	// template defines the component template.
```

Comment:
give a description about the relationship between component and shard or sharding

Reply:
> give a description about the relationship between component and shard or sharding

done